src/components/DroneStatus/MPSControlSection.tsx

```diff
@@ -35,6 +45,14 @@ export default function MPSControlSection() {
                         color="error"
                         onClick={() => {
                             setModalState(true);
+                            armDrone().then((response) => {
+                                if (response.status === 200) {
+                                    setClientSideState((prevState) => ({
+                                        ...prevState,
+                                        armed: true,
+                                    }));
+                                }
+                            });
```

Comment:
I don't think you actually need this. Because when you click this button, it brings up a modal that says "are you sure you want to arm the drone" and when you click yes on that, then it actually hits the API.